app/presenters/tree_node_builder_belongs_to_hac.rb

```diff
@@ -0,0 +1,36 @@
+class TreeNodeBuilderBelongsToHac < TreeNodeBuilder
+  def ext_management_system_node
+    super
+    @node[:select] = options[:selected].include?("#{object.class.name}_#{object[:id]}")
+  end
+
+  def host_node(object)
+    super
+    @node[:hideCheckbox] = true
+  end
+
+  def cluster_node
+    super
+    @node[:select] = options.key?(:selected) && options[:selected].include?("EmsCluster_#{object[:id]}")
+  end
+
+  def ems_folder_node
+    super
+    if object.kind_of?(Datacenter)
+      @node[:select] = options[:selected].include?("Datacenter_#{object[:id]}")
+    else
+      @node[:select] = options.key?(:selected) && options[:selected].include?("EmsFolder_#{object[:id]}")
+    end
+  end
+
+  def resource_pool_node
+    super
+    @node[:select] = options.key?(:selected) && options[:selected].include?("ResourcePool_#{object[:id]}")
+  end
+
+  def generic_node(text, image, tip = nil)
```

Comment:
Could you please do all the node overrides from above in this method? This way you would not need the one-liners in the `TreeNodeBuilder` and we can prevent some merge conflicts with [my refactoring PR](https://github.com/ManageIQ/manageiq/pull/12458). /cc @martinpovolny